src/helpers/breadcrumbHelpers.ts

```diff
@@ -0,0 +1,145 @@
+import type {
+  Breadcrumb,
+  BreadcrumbNodes,
+  BreadcrumbNode,
+  BreadcrumbTrail,
```

Comment:
Looking more at the usage, i think we should implement a `Common` component that can expose these interfaces which wraps the `Components.Breadcrumbs` from UI. That'll allow us to implement the complexity seen here that is specific to our codebase, while allowing the UI portion of the breadcrumb to be simplified by consumers